src/main/java/com/minecolonies/coremod/client/gui/WindowHutBuilder.java

```diff
@@ -242,8 +242,8 @@ private void transferItems(@NotNull final Button button)
         {
             // The itemStack size should not be greater than itemStack.getMaxStackSize, We send 1 instead
             // and use quantity for the size
-            @NotNull final ItemStack itemStack = new ItemStack(res.getItem(), 1);
-            itemStack.setTag(res.getItemStack().getTag());
+            @NotNull final ItemStack itemStack = res.getItemStack().copy();
+            itemStack.setCount(1);
```

Comment:
I feel like it might fix more bugs if we simply actually create multiple stacks with proper stack sizes instead of our NBT hacky thing

Reply:
That's what I did in this PR =P I removed the NBT hacky thing and just copyed the stack and adjusted the size.

Reply:
yeah but you're setting it to 1, it'd never go over?

Reply:
Yeah we send the amount extra.

Reply:
..... I am not sure this is the right approach here.....

Reply:
It's because we sometimes send more than the max stacksize, we have to send the amount and the stack we want to send separate.